arangod/ClusterEngine/ClusterSelectivityEstimates.cpp

```diff
@@ -30,33 +30,24 @@
 #include "Indexes/Index.h"
 #include "VocBase/LogicalCollection.h"
 
-using namespace arangodb;
+namespace arangodb {
 
 ClusterSelectivityEstimates::ClusterSelectivityEstimates(
     LogicalCollection& collection)
-    : _collection(collection), _updating(false) {}
+    : _collection(collection) {}
 
 void ClusterSelectivityEstimates::flush() {
-  // wait until we ourselves are able to set the _updating flag
-  while (_updating.load(std::memory_order_relaxed) ||
-         _updating.exchange(true, std::memory_order_acquire)) {
-    std::this_thread::yield();
-  }
-
-  auto guard = scopeGuard(
-      [this]() noexcept { _updating.store(false, std::memory_order_release); });
-
+  std::lock_guard guard{_update};
   std::atomic_store(&_data, std::shared_ptr<InternalData>());
 }
 
-IndexEstMap ClusterSelectivityEstimates::get(bool allowUpdating,
+IndexEstMap ClusterSelectivityEstimates::get(bool allowUpdate,
```

Comment:
`IndexEstMap` can be made `FlatHashMap<std::string, double>` :-)

Reply:
Done